src/components/Register.js

```diff
@@ -24,26 +26,56 @@ const Register = () => {
   const [errorVisible, setErrorVisible] = useState(false);
   const [duplicateCredentials, setDuplicateCredentials] = useState(false);
   const { handleUserRegister, isLoggedIn } = useAuthContext();
+  const [companies, setCompanies] = useState([]);
+  const [selectedValue, setSelectedValue] = useState();
+  let allCompanies = [];
+
+  useEffect(async () => {
```

Comment:
useEffect function should not be async, check this for more info - https://stackoverflow.com/questions/53332321/react-hook-warnings-for-async-function-in-useeffect-useeffect-function-must-ret